### ISSUE TITLE
Openssl 1.1.1 static builds fail to produce a viable image

### ISSUE THREAD

**Issue by tmshort** [openssl/openssl#12772]
Starting with 1.1.1c (i.e. 1.1.1b worked fine), when trying to build a non-dynamically-linked image via the `-static` option, the executable produced (at least under linux) core dumps immediately.

```
git checkout OpenSSL_1_1_1c
./config -static
make
apps/openssl version
```

This worked fine under OpenSSL 1.1.1b. The workaround is to add some options that need to be disabled:
```
git checkout OpenSSL_1_1_1c
./config -static no-shared no-pic no-threads
make
apps/openssl version
```
It appears that the fix for #8787 (commit 007213795a0e15901cbdc4558ae2a8c21d3ad9bb) broke this, based on my git bisect. As far as I can tell, the setting of some of the `$disables{}` occurs too late with that change. I have yet to figure out the solution, but I wanted to report it before I forgot about it.

**Comment by tmshort** [comment]
Although #10878 claims to have fixed this (again), it's still broken in the latest OpenSSL_1_1_1-stable.

**Comment by tmshort** [comment]
"OpenSSL 1.1.1h-dev" at commit 56456c340

**Comment by ryanwoodsmall** [comment]
1.1.1j, which contains this change, breaks previously working static+threaded+PIC builds on Linux with musl libc; this in turn breaks compiling Python with the resultant static OpenSSL libs. Static openssl bins and .a libs work fine prior to https://github.com/openssl/openssl/pull/12773 and work fine with the change reverted.

```
# uname -a
Linux ryzen9 5.8.0-43-generic #49~20.04.1-Ubuntu SMP Fri Feb 5 09:57:56 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

# gcc --version ; gcc -dumpmachine
gcc (GCC) 9.2.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

x86_64-linux-musl

# which openssl | xargs realpath | xargs file
/usr/local/crosware/software/openssl/openssl-1.1.1j/bin/openssl: ELF executable, 64-bit LSB x86-64, static, not stripped

# openssl version -a | sed 's/-I.*/.../g'
OpenSSL 1.1.1j  16 Feb 2021
built on: Wed Feb 17 06:27:18 2021 UTC
platform: linux-x86_64
options:  bn(64,64) rc4(int) des(int) idea(int) blowfish(ptr) 
compiler: x86_64-linux-musl-gcc -fPIC -pthread -m64 -Wl,-static -fPIC -fPIC -fPIC -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DZLIB -DNDEBUG -DOPENSSL_PIC -DOPENSSL_THREADS ...
OPENSSLDIR: "/usr/local/crosware/etc/ssl"
ENGINESDIR: "/usr/local/crosware/software/openssl/openssl-1.1.1j/lib/engines-1.1"
Seeding source: os-specific

# find /usr/local/crosware/software/openssl/openssl-1.1.1j/ ! -type d ! -name \*.h -exec file {} + | sort
/usr/local/crosware/software/openssl/openssl-1.1.1j/bin/c_rehash:               perl script
/usr/local/crosware/software/openssl/openssl-1.1.1j/bin/openssl:                ELF executable, 64-bit LSB x86-64, static, not stripped
/usr/local/crosware/software/openssl/openssl-1.1.1j/lib/libcrypto.a:            ar archive
/usr/local/crosware/software/openssl/openssl-1.1.1j/lib/libssl.a:               ar archive
/usr/local/crosware/software/openssl/openssl-1.1.1j/lib/pkgconfig/libcrypto.pc: ASCII text
/usr/local/crosware/software/openssl/openssl-1.1.1j/lib/pkgconfig/libssl.pc:    ASCII text
/usr/local/crosware/software/openssl/openssl-1.1.1j/lib/pkgconfig/openssl.pc:   ASCII text

# echo -e 'import ssl\nprint(ssl.OPENSSL_VERSION)' | python3 
OpenSSL 1.1.1j  16 Feb 2021

```

**Comment by tmshort** [comment]
@ryanwoodsmall you are going to run into a similar situation with the master branch (3.0) as this is basically a backport of changes done in the master branch. The change literally moved code to a place where it actually is checked. Perhaps @levitte  might have more insight?

**Comment by ryanwoodsmall** [comment]
ACK, and no worries. Easy enough to work around for now, and I know static static/threaded/PIC compilation on musl is pretty niche. Just wanted to get this posted somewhere. Thanks!

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev, barring further input
